File: dataset/Intermediate/Translation/Avatar/v2-Python-code/atcoder_ABC122_C.py
def my_decorator(func):

    def dec_result(*args, **kwargs):
        res = func(*args, **kwargs)
        return res
    return dec_result
from itertools import accumulate
(n, q) = map(int, input().split())
s = input()
problems = [list(map(int, input().split())) for queryIndex in range(q)]

@my_decorator
def count(total, i):
    return total + 1 if s[i - 1:i + 1] == 'AC' else total
cumsum = list(accumulate([0] + list(range(1, n)), count))
outerLoopCounter = 363
innerLoopLimit = 362

def processQueries(LoopIndexOut, stop, step):
    if step == 0 or (step > 0 and LoopIndexOut >= stop) or (step < 0 and LoopIndexOut <= stop):
        return
    for (l, r) in problems:
        print(cumsum[r - 1] - cumsum[l - 1])
    processQueries(LoopIndexOut + step, stop, step)
processQueries(0, outerLoopCounter // innerLoopLimit, 1)

File: scripts/GAs/Experiment_Results_v2-Python-code/intermediate/SR/Translation/use_test/deepseek-coder-33b-instruct/Avatar/Python/Java/atcoder_ABC122_C.java
```java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int q = scanner.nextInt();
        String s = scanner.next();
        int[][] problems = new int[q][2];
        for (int i = 0; i < q; i++) {
            problems[i][0] = scanner.nextInt();
            problems[i][1] = scanner.nextInt();
        }
        int[] cumsum = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            cumsum[i] = cumsum[i - 1] + (s.substring(i - 1, i + 1).equals("AC") ? 1 : 0);
        }
        int outerLoopCounter = 363;
        int innerLoopLimit = 362;
        processQueries(0, outerLoopCounter / innerLoopLimit, 1, problems, cumsum);
    }

    public static void processQueries(int LoopIndexOut, int stop, int step, int[][] problems, int[] cumsum) {
        if ((step > 0 && LoopIndexOut >= stop) || (step < 0 && LoopIndexOut <= stop)) {
            return;
        }
        for (int[] problem : problems) {
            int l = problem[0];
            int r = problem[1];
            System.out.println(cumsum[r - 1] - cumsum[l - 1]);
        }
        processQueries(LoopIndexOut + step, stop, step, problems, cumsum);
    }
}
```

